connect/src/com/telenor/connect/AbstractSdkProfile.java

```diff
@@ -166,6 +166,10 @@ protected void previewParameters(ParametersHolder parameters) {
                         && isEmpty(wellKnownConfig.getNetworkAuthenticationTargetUrls()))) {
                     parameters.add("login_hint", String.format("MCCMNC:%s", mccMnc));
         }
+
+        if (!ConnectSdk.isCellularDataNetworkConnected()) {
```

Comment:
I did that at first, hoping for the best, but it failed in my tests. Looks like ConnectSdk.getCellularNetwork() is not a reliable indicator for the availbaility of the cellular network.

For example, if you deactivate the cellular network (I switched the flight mode ON to simulate this), you won't be notified in the callback, meaning ConnectSdk.cellularNetwork will retain the old (stale) state despite the fact that the cellular network is unavailable nevertheless. You'll get your callback first when you reactivate the cellular network (e.g. by switching back the flight mode to OFF). In other words: once set to something != null, ConnectSdk.cellularNetwork will remain not null regardless of what you do.

You might need some changes in your force-HE patch to accommodate this.